buildtest/modules/spider.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+import subprocess
+
+from buildtest.tools.defaults import BUILDTEST_SPIDER_FILE, BUILDTEST_CONFIG_FILE
+from buildtest.tools.config import load_configuration
+
+
+class Spider:
+    """Class declaration of Spider class"""
+
+    def __init__(self, tree=None):
+        """Initialize method for Spider class.
+
+        :param tree: User can specify a module tree to query from spider.
+        :type tree: str
+        """
+
+        #  if user specifies a tree, then run spider command otherwise read from configuration file.
+        if tree:
+            self.tree = tree
+            spider_cmd = f"{os.getenv('LMOD_DIR')}/spider -o spider-json {self.tree}"
+
+            out = subprocess.check_output(spider_cmd, shell=True).decode("utf-8")
```

Comment:
Keeping in mind we are going to refactor BuidlTestCommand, why not use it here? And then (in the future) when we refactor, we can remove the shell=True.

Reply:
I did try using `BuildTestCommand` i had trouble getting content into a dictionary using `json.loads()` when using `get_output()` method so i gave up and used `subprocess` for now. I know this needed to be fixed.

Reply:
okay I'll add a note about this in the issue for it.